pkg/cmd/cli/describe/chaindescriber.go

```diff
@@ -0,0 +1,142 @@
+	imageapi "github.com/openshift/origin/pkg/image/api"
+	imageedges "github.com/openshift/origin/pkg/image/graph"
+	imagegraph "github.com/openshift/origin/pkg/image/graph/nodes"
+	"github.com/openshift/origin/pkg/util/parallel"
+)
+
+// ChainDescriber generates extended information about a chain of
+// dependencies of an image stream
+type ChainDescriber struct {
+	client.Interface
+}
+
+// NewChainDescriber returns a new ChainDescriber
+func NewChainDescriber(c client.Interface) *ChainDescriber {
+	return &ChainDescriber{c}
+}
+
+// MakeGraph will create the graph of all build configurations and the image streams
+// they point to via image change triggers in the provided namespace
+func (d *ChainDescriber) MakeGraph(namespace string) (osgraph.Graph, error) {
+	g := osgraph.New()
+
+	loaders := []GraphLoader{
+		NewBcLoader(namespace, d.Interface),
+		NewIsFromTriggerLoader(namespace, d.Interface, d.Interface),
+	}
+	loadingFuncs := []func() error{}
+	for _, loader := range loaders {
+		loadingFuncs = append(loadingFuncs, loader.Load)
+	}
+
+	if errs := parallel.Run(loadingFuncs...); len(errs) > 0 {
+		return g, utilerrors.NewAggregate(errs)
+	}
+
+	for _, loader := range loaders {
+		loader.AddToGraph(g)
+	}
+
+	buildedges.AddAllInputOutputEdges(g)
+	imageedges.AddAllImageStreamRefEdges(g)
+
+	return g, nil
+}
+
+func (d *ChainDescriber) Describe(namespace, name string) (string, error) {
+	g, err := d.MakeGraph(namespace)
+	if err != nil {
+		return "", err
+	}
+
+	// TODO: Traverse from 'name' and output its dependency tree
+
+	return g.String(), nil
+}
+
+// isFromTriggerLoader loads image streams found in image change triggers
+// of all build configurations in the provided namespace
+type isFromTriggerLoader struct {
+	namespace string
+	lister    client.ImageStreamsNamespacer
+	bc        *bcLoader
+	items     []imageapi.ImageStream
+}
+
+// NewIsFromTriggerLoader returns a new isFromTriggerLoader
+func NewIsFromTriggerLoader(namespace string, lister client.ImageStreamsNamespacer, bcLister client.BuildConfigsNamespacer) *isFromTriggerLoader {
+	return &isFromTriggerLoader{
+		namespace: namespace,
+		lister:    lister,
+		bc:        NewBcLoader(namespace, bcLister),
+	}
+}
+
+// Load image streams found in image change triggers of
+// all build configurations in the provided namespace
+func (l *isFromTriggerLoader) Load() error {
+	err := l.bc.Load()
+	if err != nil {
+		return err
+	}
+	return l.getStreams()
+}
+
+// AddToGraph adds all the loaded image streams and their tags
+// in the given graph
+func (l *isFromTriggerLoader) AddToGraph(g osgraph.Graph) error {
+	for i := range l.items {
+		imagegraph.EnsureImageStreamNode(g, &l.items[i])
+		imagegraph.EnsureAllImageStreamTagNodes(g, &l.items[i])
+	}
+
+	return nil
+}
+
+// getStreams iterates over a given set of build configurations
+// and extracts all the image streams which trigger a build when
+// the image stream is updated
+func (l *isFromTriggerLoader) getStreams() error {
```

Comment:
`loadStreams`.  This is a method that's mutating state, not returning a value, so `get` is the wrong verb.